### PR TITLE
option for multiple social links

### PR DESCRIPTION
**Pull request by gokatz**

### CODE REVIEW

**Comment by elwayman02** [comment]
I'm not a huge fan of treating the source repository as a "social" link. It seems to me that linking back to the root repo should have a place of prominence far above a twitter account. Can you explain the thinking behind that?

**Comment by gokatz** [comment]
Yeah, technically, repo URL is not a social link in our case. Had a little chat with @mansona, yesterday regarding this change. We are thinking of the below structure to accommodate arbitrary number of social links:
```js
{
  name: '',
  github: '',

  social: [{
    name: 'twitter',
    link: 'https://twitter.com/emberjs'
  }, {
    ...
  }]
}
```
Though there is another question in this approach. There are multiple GIT hosting providers like GitHub, GitLab (#9), bitbucket etc., If we maintain the repo URL outside the social link, should we allow keys for other providers on the root as well? If so, should we maintain their official logos? or can we use the generic GIT logo instead of GitHub's?

**Comment by mansona** [comment]
Yea that's a good point @elwayman02 🤔 the way that we've been solving this idea of prominence in the ember-styleguide is that we just link to the repo in the index file: https://ember-styleguide.netlify.app/#contributingtotheproject

This API that we're discussing is to power the low-prominence "social links" in the bottom right of the current default template 

Image 2020-05-07 at 8 55 01 a m

Did you have an idea how adding more codification of the repository in the fieldGuide config might display it differently?

**Comment by elwayman02** [comment]
I would be fine renaming the `github` config to `repository` and then looking at the base of that url to determine the provider. We could then infer the correct logo for a couple major providers and fallback to the git logo for any other url.

IMO linking back to the source repo for the docs we're browsing should be much more prominent than we have it now. It's fairly common to want to go from the docs to the source code. My point is that the repo itself is, in this way, far more prominent of a detail, so we may at some point want to update the design of field-guide to put that link in a more prominent position of the page.

**Comment by gokatz** [comment]
the `repository` key sounds fine. Finding the provider name with the URL host will work for most case. However, in my case, as we are having a self-hosted GitLab instance, the fallback generic git logo will be used. But I prefer a GitLab logo. Hence we can expose the name of the logo used so that the host application can override the logo if needed.

**Comment by gokatz** [comment]
wrt to the repo link, can we have it fixed somewhere in the sidebar?

Screenshot 2020-05-10 at 11 28 31 AM

**Comment by elwayman02** [comment]
I really like the idea of sticking it at the bottom of the sidebar, which for many smaller projects is going to have a lot of dead space anyway

File: blueprints/field-guide/index.js [hunk 1]
```diff
@@ -43,7 +43,10 @@ module.exports = {
     applyConfig(this.project, 'field-guide', {
       name: 'Product Name',
       copyright: 'This is the default copyright string - update before publishing',
-      github: 'https://github.com/empress/field-guide?update-with-your-repo-url'
+      social: {
+        github: 'https://github.com/ember-learn/ember-styleguide',
```
Comment (mansona) [inline blueprints/field-guide/index.js:47]:
```suggestion
        github: 'https://github.com/ember-learn/ember-styleguide?update-with-your-repo-url',
```

File: blueprints/field-guide/index.js [hunk 2]
```diff
@@ -43,7 +43,10 @@ module.exports = {
...
+        twitter: 'https://twitter.com/emberjs'
```
Comment (mansona) [inline blueprints/field-guide/index.js:48]:
```suggestion
        twitter: 'https://twitter.com/emberjs?replace-with-your-own-twitter'
```

**Comment by mansona** [comment]
So I'm happy with where these conversations are going 👍 but it all seems like an "as well as" conversation rather than an "instead of" conversation 🤔 

For sure we should link back to the origin repo and some of the suggestions are good for sure 👍 but we have some more thinking and discussion before they are going to be ready as a plan other than what @gokatz is proposing in this PR

@gokatz I recommend adding the slight changes that I have suggested and implementing the "fallback" that we discussed before (i.e. make the github logo show up if you already have `github` defined in your config) but skip the deprecation message that we discussed 👍 that can happen as part of the other issue/PR whenever it comes.

Does that make sense?

**Comment by elwayman02** [comment]
👍 on array for arbitrary number of social links, by the way

**Pull request merged**